specs/plasma.md

```diff
@@ -0,0 +1,114 @@
+can challenge a commitment `challenge_window` and the period during which any party
+can submit the input data to resolve the challenge `resolve_window`.
+If a challenge is not resolved by the time `resolve_window` of L1 blocks have passed,
+chain derivation must be reset, omitting the input data when rederiving therefore causing a reorg.
+
+## DA Storage
+
+Input data is uploaded to the storage layer via plain HTTP calls to a stateless DA storage service.
+This service is horizontally scalable and concerned with replicating the data across prefered storage layers
+such as IPFS or any S3 compatible storage. Input data is content addressed by its hash in the request url
+so responses are easily cachable.
+
+Any DA provider can implement the following endpoints to receive and serve input data:
+
+- `POST /put/<hex_encoded_commitment> -H "Content-Type: application/octet-stream"`
+- `GET /get/<hex_encoded_commitment> -H "Content-Type: application/octet-stream"`
+
+## Input Commitment
+
+The batching and compression of input data remain unchanged. When a batch is ready
+to be submitted to the inbox address, the data is uploaded to the DA storage layer instead, and a
+commitment (keccak256 hash) is submitted as the bacher inbox transaction call data.
+The batcher will not submit a commitment onchain unless input data was successfully stored on the service.
+
+## Data Availability Challenge Contract
+
+Data availability is guaranteed via a permissionless challenge contract on the L1 chain.
+Users have a set number of L1 blocks (AKA `challenge_window`) during which they are able to call
+the `challenge` method of the contract with the following inputs:
+
+- A Commitment type (i.e. keccak256)
+- Commitment bytes
+- The L1 block number in which it was included.
+
+Users with access to the input data then have another window of L1 blocks (AKA `resolve_window`)
+during which they can submit it as calldata to the chain by calling the `resolve` method of the contract.
+If the data is not included onchain by the time the resolve window is elapsed, derivation of the L2 canonical chain
+will omit the input data, hence any transaction included in the input data will be dropped
+causing a reorg of any L2 chain that previously had included that transaction.
+
+In order to challenge a commitment, users must deposit a bond amount where `bond >= resolve_tx_gas_cost`.
+If the gas cost of resolving the challenge was lower than the bond, the difference is reimbursed to the challenger
+and the rest of the bond is burnt. If the challenge is not resolved in time and expired,
+the bond is returned and can be withdrawn by the challenger or used to challenge another commitment.
+
+The state of all challenges can be read from the contract state or by syncing contract events.
+`challenge_window` and `resolve_window` are constant values that currently cannot be changed
+unless the contract is upgraded. A dynamic window mechanism may be explored in the future.
+
```

Comment:
When the data is posted to L1 and made available, would the `op-node` continue its sync using that data? Without this feature, chains would halt in practice but users would be able to see that the data was made available